---
output: 
    github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

<!-- badges: start -->
[![R-CMD-check](https://github.com/jleechung/Banksy/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/jleechung/Banksy/actions/workflows/R-CMD-check.yaml)
[![codecov](https://codecov.io/gh/jleechung/Banksy/branch/main/graph/badge.svg?token=OZZK4EDVH9)](https://codecov.io/gh/jleechung/Banksy)
[![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://github.com/jleechung/Banksy/graphs/commit-activity)
<!-- badges: end -->

## Overview

BANKSY is a method for clustering spatial transcriptomic data by augmenting the
transcriptomic profile of each cell with an average of the transcriptomes of its
spatial neighbors. By incorporating neighborhood information for clustering, 
BANKSY is able to

- improve cell-type assignment in noisy data
- distinguish subtly different cell-types stratified by microenvironment
- identify spatial zones sharing the same microenvironment

BANKSY is applicable to a wide array of spatial technologies (e.g. 10x Visium, 
Slide-seq, MERFISH) and scales well to large datasets. For more details, check 
out:

- the [preprint](https://www.biorxiv.org/content/10.1101/2022.04.14.488259v1),
- a [tweetorial](https://twitter.com/vipul1891/status/1515323372535644166?s=20&t=Bc6rz8VeWWptF67FejGYfQ) on BANKSY, 
- and a [Python version](https://github.com/prabhakarlab/Banksy_py) of this package. 

## Installation

The *Banksy* package can be installed via `remotes`:

```{r, eval=F}
remotes::install_github("prabhakarlab/Banksy", dependencies = TRUE)
```

Installation should take less than three minutes. 

**Known installation issues**

1. Installation of `leidenAlg` has non-zero exit status

- Refer to the [package website](https://github.com/kharchenkolab/leidenAlg#installation) for 
*leidenAlg* installation details. Otherwise, users may also install a separate branch
of *Banksy* with 

```{r, eval=F}
remotes::install_github("prabhakarlab/Banksy@feat-igraph-leiden")
```

## Documentation

Detailed description of *Banksy* functionality and example analyses are available
at the [package webpage](https://prabhakarlab.github.io/Banksy/). 

*Banksy* comes installed with [documentation](https://prabhakarlab.github.io/Banksy/reference/index.html) of main functions and their 
usage, along with several vignettes which detail different use cases:

* [Working with Banksy objects](https://prabhakarlab.github.io/Banksy/articles/banksy-object.html):
  Introduction to the *BanksyObject* class which serves as a container for 
  *Banksy*.  
  
* [Mouse hippocampus VeraFISH dataset](https://prabhakarlab.github.io/Banksy/articles/hippocampus-analysis.html):
  Illustrates a grid search of parameters which best cluster cells.  

* [Human dorsolateral prefrontal cortex 10x Visium dataset](https://prabhakarlab.github.io/Banksy/articles/dlpfc-analysis.html):
  Illustrates analysis of multiple spatial transcriptomic datasets.

* [Mouse hypothalamus MERFISH dataset](https://prabhakarlab.github.io/Banksy/articles/hypothalamus-analysis.html):
  Illustrates visualization functionality with a dataset with 3 spatial dimensions.
  
* [Interoperability with SingleCellExperiment](https://prabhakarlab.github.io/Banksy/articles/single-cell-exp.html):
  Illustrates BANKSY interoperability with Bioconductor
  [SingleCellExperiment](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html)
  framework for interfacing with packages like [scran](https://bioconductor.org/packages/release/bioc/html/scran.html) 
  or [scater](https://bioconductor.org/packages/release/bioc/html/scater.html).

* [Figure 4 data analysis](https://prabhakarlab.github.io/Banksy/articles/Fig4_vignette.html):
  Shows how the results shown in Fig. 4 of the paper were generated. 
  
*Banksy* is also interoperable with [Seurat](https://satijalab.org/seurat/) via
*SeuratWrappers*. Documentation on how to run BANKSY on Seurat objects can be 
found [here](https://github.com/satijalab/seurat-wrappers/blob/master/docs/banksy.md). 
For installation of *SeuratWrappers* with the latest BANKSY version, run
```{r, eval=F}
remotes::install_github('jleechung/seurat-wrappers')
```

## Quick start

*Banksy* takes as input an expression matrix and cell centroids. Example 
datasets are provided with the package:

```{r, eval = T, include=FALSE}
start.time <- Sys.time()
```

```{r, eval = T}
library(Banksy)

data(hippocampus)
expr <- hippocampus$expression
locs <- hippocampus$locations
```

The gene expression matrix for cells should be a `matrix`:

```{r, eval = T}
class(expr)
head(expr[,1:5])
```

while cell locations should be supplied as a `data.frame`:

```{r, eval = T}
class(locs)
head(locs)
```

Next, create a *BanksyObject* with the expression matrix and cell locations.

```{r, eval = T}
bank <- BanksyObject(own.expr = expr,
                     cell.locs = locs)

head(bank)
```

Apply basic QC by keeping only cells with total counts within the 5th and 98th 
percentile:

```{r, eval = T}
bank <- SubsetBanksy(bank, metadata = nCount > quantile(nCount, 0.05) &
                                      nCount < quantile(nCount, 0.98))
```

We first normalize the expression matrix, compute the neighbor matrix, and 
scale the resulting matrices.

```{r, eval = T}
bank <- NormalizeBanksy(bank)
bank <- ComputeBanksy(bank, k_geom = c(15, 30))
bank <- ScaleBanksy(bank)
```

Run PCA on the BANKSY matrix for `lambda=0` (no spatial information) 
and `lambda=0.2`. 

```{r, eval = T}
bank <- RunBanksyPCA(bank, lambda = c(0, 0.2))
```

Next, we obtain cluster assignments using graph-based clustering with the Leiden
algorithm on the first 20 PCs. 

* `resolution`. Leiden clustering resolution.  

```{r, eval = T}
set.seed(42)
bank <- ClusterBanksy(bank, lambda = c(0, 0.2), pca = TRUE, npcs = 20,
                      method = 'leiden', resolution = 1.2)
```

Different clustering runs can be harmonised with `ConnectClusters`:

```{r, eval = T}
bank <- ConnectClusters(bank, map.to = clust.names(bank)[1])
``` 

Visualise the clustering output for non-spatial clustering (`lambda=0`) and 
BANKSY clustering (`lambda=0.2`). 

```{r, eval = T, fig.height=8, fig.width=18}
features <- clust.names(bank)
feature.types <- rep('discrete', 2)
main <- c('Non-spatial', 'BANKSY')
plotSpatialFeatures(bank, by = features, type = feature.types, main = main, 
                    pt.size = 1.5, main.size = 15, nrow = 1, ncol = 2)
```

For clarity, we can visualise each of the clusters separately with `wrap = TRUE`:

```{r, eval = T, fig.height=8, fig.width=18}
plotSpatialFeatures(bank, by = features, type = feature.types, main = main, 
                    pt.size = 1, main.size = 15, nrow = 1, ncol = 2, 
                    wrap = TRUE)
``` 

<details>
    <summary>Runtime for analysis</summary>

```{r, eval = T, echo=FALSE}
Sys.time() - start.time
```

</details>

<details>
    <summary>Session information</summary>

```{r, sess}
sessionInfo()
```

</details>

## Contributing

Bug reports, questions, request for enhancements or other contributions can be 
raised at the [issue page](https://github.com/prabhakarlab/Banksy/issues). 
