---
title: "Interoperability with SingleCellExperiment"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Interoperability with SingleCellExperiment}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Here, we demonstrate interoperability between *Banksy* and 
*SingleCellExperiment* with a mouse VeraFISH dataset.  

## Loading the data

The dataset comprises gene expression for 10,944 cells and 120 genes in 2 
spatial dimensions. See `?Banksy::hippocampus` for more details. We load 
*Bioconductor* packages for single-cell analyses, *scater* and *scran*, and
create the *SingleCellExperiment* object. 

```{r, message=FALSE, warning=FALSE}
library(Banksy)
library(scater)
library(scran)
library(gridExtra)

data(hippocampus)

# Create SCE
sce <- SingleCellExperiment(
    assays = list(counts = hippocampus$expression), 
    colData = hippocampus$locations)
sce <- scater::logNormCounts(sce)
```

## Running BANKSY

We convert the *SingleCellExperiment* object to a *BanksyObject*, and run the
BANKSY pipeline. In summary, we run BANKSY with `lam=0` corresponding to 
non-spatial clustering, and `lam=0.3` which incorporates spatial information. 
We compute 20 PCs, construct a shared nearest neighbor graph and cluster the
graph with Leiden clustering. 

```{r}
# Convert to BanksyObject
bank <- asBanksyObject(sce, expr.assay = 'counts', 
                       coord.colnames = c('sdimx', 'sdimy'))
# Run pipeline
bank <- NormalizeBanksy(bank, normFactor = 100)
bank <- ComputeBanksy(bank, k_geom = 10, spatialMode = 'kNN_r')
bank <- ScaleBanksy(bank)
bank <- RunPCA(bank, lambda = c(0,0.3), npcs = 20)
bank <- RunUMAP(bank, lambda = c(0,0.3), pca = TRUE, npcs = 20)
set.seed(42)
bank <- ClusterBanksy(bank, lambda = c(0,0.3), npcs = 20, 
                      method = 'leiden', k.neighbors = 50, resolution = 1)
bank <- ConnectClusters(bank, map.to = clust.names(bank)[1])
```

## Appending BANKSY output

Output from the BANKSY run can be added to the original *SingleCellExperiment*
object. Here, we add the cluster labels and the PCA and UMAP cell embeddings. 

```{r}
# Add data to SCE
sce$non_spatial <- factor(meta.data(bank)$clust_lam0_k50_res1)
sce$BANKSY <- factor(meta.data(bank)$clust_lam0.3_k50_res1)
reducedDims(sce) <- list(UMAP_nsp = reduction(bank)$umap_0,
                         UMAP_bky = reduction(bank)$umap_0.3,
                         PCA_nsp = reduction(bank)$pca_0$x,
                         PCA_bky = reduction(bank)$pca_0.3$x)
```

Visualise the cells in UMAP and spatial dimensions:

```{r, fig.align='center', fig.width=9, fig.height=4}
grid.arrange(
    plotColData(sce, x = 'sdimx', y = 'sdimy', colour_by = 'non_spatial',
                point_size = 0.05) + facet_wrap(~ colour_by),
    plotColData(sce, x = 'sdimx', y = 'sdimy', colour_by = 'BANKSY',
                point_size = 0.05) + facet_wrap(~ colour_by),
    ncol = 2)
grid.arrange(
    plotReducedDim(sce, dimred = 'UMAP_nsp', colour_by = 'non_spatial', 
                   point_size = 0.05),
    plotReducedDim(sce, dimred = 'UMAP_bky', colour_by = 'BANKSY', 
                   point_size = 0.05),
    ncol = 2)
```

We can find markers for each BANKSY cluster with *scran*:

```{r, eval=FALSE, fig.align='center', fig.width=9, fig.height=7}
colLabels(sce) <- sce$BANKSY
marker.info <- scoreMarkers(sce, colLabels(sce))
features <- sapply(marker.info, function(m) {
    m <- m[order(m$mean.AUC, decreasing = TRUE),]
    head(rownames(m), n = 1)
})
plotExpression(sce, features = features, x = 'BANKSY', colour_by = 'BANKSY',
               point_size = 0.1, ncol = 3)

```

## Session information

<details>

```{r, sess}
sessionInfo()
```

</details>
