<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Mouse Hippocampus VeraFISH analysis â€¢ Banksy</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Mouse Hippocampus VeraFISH analysis">
<meta property="og:description" content="Banksy">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Banksy</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Functions and data</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/banksy-object.html">Working with BanksyObjects</a>
    </li>
    <li>
      <a href="../articles/hippocampus-analysis.html">Choosing clustering parameters (Mouse hippocampus VeraFISH)</a>
    </li>
    <li>
      <a href="../articles/dlpfc-analysis.html">Analysing multiple datasets (Human DLPFC 10x Visium)</a>
    </li>
    <li>
      <a href="../articles/hypothalamus-analysis.html">Visualization (Mouse hypothalamus MERFISH)</a>
    </li>
    <li>
      <a href="../articles/single-cell-exp.html">SingleCellExperiment interoperability</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/prabhakarlab/Banksy/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Mouse Hippocampus VeraFISH analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/prabhakarlab/Banksy/blob/HEAD/vignettes/hippocampus-analysis.Rmd" class="external-link"><code>vignettes/hippocampus-analysis.Rmd</code></a></small>
      <div class="hidden name"><code>hippocampus-analysis.Rmd</code></div>

    </div>

    
    
<p>Here, we demonstrate a grid search of clustering parameters with a
mouse hippocampus VeraFISH dataset. <em>BANKSY</em> currently provides
four algorithms for clustering the BANKSY matrix with
<em>ClusterBanksy</em>: Leiden (default), Louvain, k-means, and
model-based clustering. In this vignette, we run only Leiden clustering.
See <code><a href="../reference/ClusterBanksy.html">?ClusterBanksy</a></code> for more details on the parameters for
different clustering methods.</p>
<div class="section level2">
<h2 id="loading-the-data">Loading the data<a class="anchor" aria-label="anchor" href="#loading-the-data"></a>
</h2>
<p>First, we construct the <em>BanksyObject</em>. The dataset comprises
gene expression for 10,944 cells and 120 genes in 2 spatial dimensions.
See <code><a href="../reference/hippocampus.html">?Banksy::hippocampus</a></code> for more details.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/prabhakarlab/Banksy" class="external-link">Banksy</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hippocampus</span><span class="op">)</span></span>
<span><span class="va">expr</span> <span class="op">&lt;-</span> <span class="va">hippocampus</span><span class="op">$</span><span class="va">expression</span></span>
<span><span class="va">locs</span> <span class="op">&lt;-</span> <span class="va">hippocampus</span><span class="op">$</span><span class="va">locations</span></span>
<span></span>
<span><span class="co"># Store total counts and number of expressed genes</span></span>
<span><span class="va">total_count</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span></span>
<span><span class="va">num_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">expr</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>total_count <span class="op">=</span> <span class="va">total_count</span>, num_genes <span class="op">=</span> <span class="va">num_genes</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Construct BanksyObject</span></span>
<span><span class="va">bank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BanksyObject-class.html">BanksyObject</a></span><span class="op">(</span>own.expr <span class="op">=</span> <span class="va">expr</span>, cell.locs <span class="op">=</span> <span class="va">locs</span>, meta.data <span class="op">=</span> <span class="va">meta</span><span class="op">)</span></span>
<span><span class="va">bank</span></span>
<span><span class="co">#&gt; Object of class BanksyObject </span></span>
<span><span class="co">#&gt; Assay with 10994 cells 120 features</span></span>
<span><span class="co">#&gt; Spatial dimensions: sdimx sdimy </span></span>
<span><span class="co">#&gt; Metadata names: cell_ID nCount NODG </span></span>
<span><span class="co">#&gt; Dimension reductions:</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="parameters">Parameters<a class="anchor" aria-label="anchor" href="#parameters"></a>
</h2>
<p>The <code>lambda</code> parameter is a mixing parameter in
<code>[0,1]</code> which determines how much spatial information is
incorporated for clustering. With smaller values of <code>lambda</code>
(e.g.Â <code>[0,0.25]</code>), BANKY operates in <em>cell-typing</em>
mode, while at higher levels of <code>lambda</code>, BANKSY operates in
<em>zone-finding</em> mode. See the pre-print for more details.</p>
<p>Leiden graph-based clustering admits two parameters:
<code>k.neighbors</code> and <code>resolution</code>.
<code>k.neighbors</code> determines the number of k nearest neighbors
used to construct the shared nearest neighbors graph. Leiden clustering
is then performed on the resultant graph with resolution
<code>resolution</code>.</p>
<p>We first process the data for clustering:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SubsetBanksy.html">SubsetBanksy</a></span><span class="op">(</span><span class="va">bank</span>, metadata <span class="op">=</span> <span class="va">total_count</span> <span class="op">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">total_count</span>, <span class="fl">0.05</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>                                      <span class="va">total_count</span> <span class="op">&lt;</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">total_count</span>, <span class="fl">0.98</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NormalizeBanksy.html">NormalizeBanksy</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span></span>
<span><span class="va">bank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ComputeBanksy.html">ComputeBanksy</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span></span>
<span><span class="va">bank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ScaleBanksy.html">ScaleBanksy</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="grid-search">Grid search<a class="anchor" aria-label="anchor" href="#grid-search"></a>
</h2>
<p>Before clustering, we define a parameter space. Here, weâ€™ll explore
different combinations of <code>lambda</code> and
<code>resolution</code> while fixing <code>kneighbours</code>. We first
run PCA on the BANKSY matrix with 20 PCs for each <code>lambda</code>.
Clustering is then performed on the PCs (by default):</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.8</span>, <span class="fl">1.2</span><span class="op">)</span></span>
<span><span class="va">knbr</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span></span>
<span><span class="va">bank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunBanksyPCA.html">RunBanksyPCA</a></span><span class="op">(</span><span class="va">bank</span>, lambda <span class="op">=</span> <span class="va">lam</span>, npcs <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt; Running PCA for M=0 lambda=0</span></span>
<span><span class="co">#&gt; BANKSY matrix with own.expr, F0</span></span>
<span><span class="co">#&gt; Squared lambdas: 1, 0</span></span>
<span><span class="co">#&gt; Running PCA for M=0 lambda=0.2</span></span>
<span><span class="co">#&gt; BANKSY matrix with own.expr, F0</span></span>
<span><span class="co">#&gt; Squared lambdas: 0.8, 0.2</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span>
<span><span class="va">bank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ClusterBanksy.html">ClusterBanksy</a></span><span class="op">(</span><span class="va">bank</span>, lambda <span class="op">=</span> <span class="va">lam</span>, pca <span class="op">=</span> <span class="cn">TRUE</span>, npcs <span class="op">=</span> <span class="fl">20</span>, </span>
<span>                      method <span class="op">=</span> <span class="st">'leiden'</span>, k.neighbors <span class="op">=</span> <span class="va">knbr</span>, resolution <span class="op">=</span> <span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using 1 core. Consider parallelising with the num.cores argument</span></span></code></pre></div>
<p>This populates the <code>meta.data</code> slot of the
<em>BanksyObject</em> with cluster labels for each combination of
parameters:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/BanksyObject-class.html">meta.data</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;             cell_ID nCount NODG clust_M0_lam0_k50_res0.8</span></span>
<span><span class="co">#&gt; cell_1276 cell_1276    266   51                        8</span></span>
<span><span class="co">#&gt; cell_691   cell_691    132   36                        4</span></span>
<span><span class="co">#&gt; cell_396   cell_396     95   27                        1</span></span>
<span><span class="co">#&gt; cell_68     cell_68    579   72                        8</span></span>
<span><span class="co">#&gt; cell_6954 cell_6954    116   29                        7</span></span>
<span><span class="co">#&gt; cell_7074 cell_7074     28   17                        6</span></span>
<span><span class="co">#&gt;           clust_M0_lam0_k50_res1.2 clust_M0_lam0.2_k50_res0.8</span></span>
<span><span class="co">#&gt; cell_1276                        9                         13</span></span>
<span><span class="co">#&gt; cell_691                         3                          8</span></span>
<span><span class="co">#&gt; cell_396                         7                         13</span></span>
<span><span class="co">#&gt; cell_68                          9                         13</span></span>
<span><span class="co">#&gt; cell_6954                        8                          7</span></span>
<span><span class="co">#&gt; cell_7074                        6                          7</span></span>
<span><span class="co">#&gt;           clust_M0_lam0.2_k50_res1.2</span></span>
<span><span class="co">#&gt; cell_1276                         16</span></span>
<span><span class="co">#&gt; cell_691                           7</span></span>
<span><span class="co">#&gt; cell_396                          16</span></span>
<span><span class="co">#&gt; cell_68                           16</span></span>
<span><span class="co">#&gt; cell_6954                          6</span></span>
<span><span class="co">#&gt; cell_7074                          6</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cluster-similarity">Cluster similarity<a class="anchor" aria-label="anchor" href="#cluster-similarity"></a>
</h2>
<p>To compare clustering labels from different parameter combinations,
the package implements the <em>getARI</em> and <em>plotARI</em>
functions. These compute the adjusted Rand index (ARI) for all pairs of
clusters to assess the similarity of clustering labels. Observe that
clusters with the same <code>lambda</code> share more similarity:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getARI.html">getARI</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span></span>
<span><span class="co">#&gt;                            clust_M0_lam0_k50_res0.8 clust_M0_lam0_k50_res1.2</span></span>
<span><span class="co">#&gt; clust_M0_lam0_k50_res0.8                      1.000                    0.893</span></span>
<span><span class="co">#&gt; clust_M0_lam0_k50_res1.2                      0.893                    1.000</span></span>
<span><span class="co">#&gt; clust_M0_lam0.2_k50_res0.8                    0.781                    0.731</span></span>
<span><span class="co">#&gt; clust_M0_lam0.2_k50_res1.2                    0.698                    0.684</span></span>
<span><span class="co">#&gt;                            clust_M0_lam0.2_k50_res0.8</span></span>
<span><span class="co">#&gt; clust_M0_lam0_k50_res0.8                        0.781</span></span>
<span><span class="co">#&gt; clust_M0_lam0_k50_res1.2                        0.731</span></span>
<span><span class="co">#&gt; clust_M0_lam0.2_k50_res0.8                      1.000</span></span>
<span><span class="co">#&gt; clust_M0_lam0.2_k50_res1.2                      0.843</span></span>
<span><span class="co">#&gt;                            clust_M0_lam0.2_k50_res1.2</span></span>
<span><span class="co">#&gt; clust_M0_lam0_k50_res0.8                        0.698</span></span>
<span><span class="co">#&gt; clust_M0_lam0_k50_res1.2                        0.684</span></span>
<span><span class="co">#&gt; clust_M0_lam0.2_k50_res0.8                      0.843</span></span>
<span><span class="co">#&gt; clust_M0_lam0.2_k50_res1.2                      1.000</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotARI.html">plotARI</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span> </span></code></pre></div>
<p><img src="hippocampus-analysis_files/figure-html/runari-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="connecting-clusters">Connecting clusters<a class="anchor" aria-label="anchor" href="#connecting-clusters"></a>
</h2>
<p>To visually compare between clusters obtained with different
combination of parameters (a parameter run), the package implements
<em>ConnectClusters</em> which performs a mapping to harmonise cluster
labels between different parameter runs.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ConnectClusters.html">ConnectClusters</a></span><span class="op">(</span><span class="va">bank</span>, map.to <span class="op">=</span> <span class="st">'clust_M0_lam0.2_k50_res1.2'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/BanksyObject-class.html">meta.data</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;             cell_ID nCount NODG clust_M0_lam0_k50_res0.8</span></span>
<span><span class="co">#&gt; cell_1276 cell_1276    266   51                       12</span></span>
<span><span class="co">#&gt; cell_691   cell_691    132   36                        7</span></span>
<span><span class="co">#&gt; cell_396   cell_396     95   27                        2</span></span>
<span><span class="co">#&gt; cell_68     cell_68    579   72                       12</span></span>
<span><span class="co">#&gt; cell_6954 cell_6954    116   29                        6</span></span>
<span><span class="co">#&gt; cell_7074 cell_7074     28   17                        5</span></span>
<span><span class="co">#&gt;           clust_M0_lam0_k50_res1.2 clust_M0_lam0.2_k50_res0.8</span></span>
<span><span class="co">#&gt; cell_1276                       12                         16</span></span>
<span><span class="co">#&gt; cell_691                         7                          7</span></span>
<span><span class="co">#&gt; cell_396                         2                         16</span></span>
<span><span class="co">#&gt; cell_68                         12                         16</span></span>
<span><span class="co">#&gt; cell_6954                        6                          6</span></span>
<span><span class="co">#&gt; cell_7074                        5                          6</span></span>
<span><span class="co">#&gt;           clust_M0_lam0.2_k50_res1.2</span></span>
<span><span class="co">#&gt; cell_1276                         16</span></span>
<span><span class="co">#&gt; cell_691                           7</span></span>
<span><span class="co">#&gt; cell_396                          16</span></span>
<span><span class="co">#&gt; cell_68                           16</span></span>
<span><span class="co">#&gt; cell_6954                          6</span></span>
<span><span class="co">#&gt; cell_7074                          6</span></span></code></pre></div>
<p>This updates the <code>meta.data</code> slot with new cluster labels
that can be used for visualisation. We can visualise connected output as
follows.</p>
<p>First, obtain cluster names:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cnms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BanksyObject-class.html">clust.names</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span></span></code></pre></div>
<p>Then, visualise the cluster runs:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialFeatures.html">plotSpatialFeatures</a></span><span class="op">(</span><span class="va">bank</span>, by <span class="op">=</span> <span class="va">cnms</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">'discrete'</span>, <span class="fl">4</span><span class="op">)</span>, </span>
<span>                    nrow <span class="op">=</span> <span class="fl">2</span>, ncol <span class="op">=</span> <span class="fl">2</span>, main <span class="op">=</span> <span class="va">cnms</span>, main.size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="hippocampus-analysis_files/figure-html/out-1.png" width="960"></p>
<p>We visualise the differences between clusters obtained with
non-spatial clustering (<code>lam=0</code>) and BANKSY in cell-typing
mode (<code>lam=0.3</code>) more finely by splitting the clusters with
<code>wrap = TRUE</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotSpatial.html">plotSpatial</a></span><span class="op">(</span><span class="va">bank</span>, by <span class="op">=</span> <span class="va">cnms</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">'discrete'</span>, </span>
<span>            main <span class="op">=</span> <span class="va">cnms</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, main.size <span class="op">=</span> <span class="fl">10</span>, pt.size <span class="op">=</span> <span class="fl">0.1</span>, wrap <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotSpatial.html">plotSpatial</a></span><span class="op">(</span><span class="va">bank</span>, by <span class="op">=</span> <span class="va">cnms</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">'discrete'</span>, </span>
<span>            main <span class="op">=</span> <span class="va">cnms</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>, main.size <span class="op">=</span> <span class="fl">10</span>, pt.size <span class="op">=</span> <span class="fl">0.1</span>, wrap <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="hippocampus-analysis_files/figure-html/wrap-1.png" width="1152" style="display: block; margin: auto;"></p>
<p>In addition, one can also visualise differences in the clustering
labels with alluvial plots:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotAlluvia.html">plotAlluvia</a></span><span class="op">(</span><span class="va">bank</span><span class="op">)</span></span></code></pre></div>
<p><img src="hippocampus-analysis_files/figure-html/alluv-1.png" width="672" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<details><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.2.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/London</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] gridExtra_2.3 ggplot2_3.4.4 Banksy_0.1.5 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] bitops_1.0-7                rlang_1.1.3                </span></span>
<span><span class="co">#&gt;   [3] magrittr_2.0.3              clue_0.3-65                </span></span>
<span><span class="co">#&gt;   [5] GetoptLong_1.0.5            matrixStats_1.2.0          </span></span>
<span><span class="co">#&gt;   [7] compiler_4.3.2              sccore_1.0.4               </span></span>
<span><span class="co">#&gt;   [9] reshape2_1.4.4              png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [11] systemfonts_1.0.5           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [13] maps_3.4.2                  ggalluvial_0.12.5          </span></span>
<span><span class="co">#&gt;  [15] stringr_1.5.1               pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [17] shape_1.4.6                 crayon_1.5.2               </span></span>
<span><span class="co">#&gt;  [19] fastmap_1.1.1               XVector_0.42.0             </span></span>
<span><span class="co">#&gt;  [21] labeling_0.4.3              utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [23] rmarkdown_2.25              ragg_1.2.7                 </span></span>
<span><span class="co">#&gt;  [25] purrr_1.0.2                 xfun_0.42                  </span></span>
<span><span class="co">#&gt;  [27] zlibbioc_1.48.0             cachem_1.0.8               </span></span>
<span><span class="co">#&gt;  [29] pals_1.8                    GenomeInfoDb_1.38.6        </span></span>
<span><span class="co">#&gt;  [31] jsonlite_1.8.8              progress_1.2.3             </span></span>
<span><span class="co">#&gt;  [33] highr_0.10                  DelayedArray_0.28.0        </span></span>
<span><span class="co">#&gt;  [35] prettyunits_1.2.0           irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt;  [37] parallel_4.3.2              cluster_2.1.6              </span></span>
<span><span class="co">#&gt;  [39] R6_2.5.1                    bslib_0.6.1                </span></span>
<span><span class="co">#&gt;  [41] stringi_1.8.3               RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [43] leidenAlg_1.1.2             GenomicRanges_1.54.1       </span></span>
<span><span class="co">#&gt;  [45] jquerylib_0.1.4             Rcpp_1.0.12                </span></span>
<span><span class="co">#&gt;  [47] SummarizedExperiment_1.32.0 iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [49] knitr_1.45                  IRanges_2.36.0             </span></span>
<span><span class="co">#&gt;  [51] Matrix_1.6-5                igraph_2.0.1.1             </span></span>
<span><span class="co">#&gt;  [53] tidyselect_1.2.0            rstudioapi_0.15.0          </span></span>
<span><span class="co">#&gt;  [55] dichromat_2.0-0.1           abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [57] yaml_2.3.8                  doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [59] codetools_0.2-19            lattice_0.22-5             </span></span>
<span><span class="co">#&gt;  [61] tibble_3.2.1                plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [63] withr_3.0.0                 Biobase_2.62.0             </span></span>
<span><span class="co">#&gt;  [65] evaluate_0.23               desc_1.4.3                 </span></span>
<span><span class="co">#&gt;  [67] circlize_0.4.15             mclust_6.0.1               </span></span>
<span><span class="co">#&gt;  [69] pillar_1.9.0                MatrixGenerics_1.14.0      </span></span>
<span><span class="co">#&gt;  [71] foreach_1.5.2               stats4_4.3.2               </span></span>
<span><span class="co">#&gt;  [73] generics_0.1.3              dbscan_1.1-12              </span></span>
<span><span class="co">#&gt;  [75] RCurl_1.98-1.14             S4Vectors_0.40.2           </span></span>
<span><span class="co">#&gt;  [77] hms_1.1.3                   munsell_0.5.0              </span></span>
<span><span class="co">#&gt;  [79] scales_1.3.0                glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [81] mapproj_1.2.11              tools_4.3.2                </span></span>
<span><span class="co">#&gt;  [83] data.table_1.15.0           fs_1.6.3                   </span></span>
<span><span class="co">#&gt;  [85] grid_4.3.2                  tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [87] colorspace_2.1-0            GenomeInfoDbData_1.2.11    </span></span>
<span><span class="co">#&gt;  [89] RcppHungarian_0.3           cli_3.6.2                  </span></span>
<span><span class="co">#&gt;  [91] textshaping_0.3.7           fansi_1.0.6                </span></span>
<span><span class="co">#&gt;  [93] S4Arrays_1.2.0              ComplexHeatmap_2.18.0      </span></span>
<span><span class="co">#&gt;  [95] dplyr_1.1.4                 uwot_0.1.16                </span></span>
<span><span class="co">#&gt;  [97] gtable_0.3.4                sass_0.4.8                 </span></span>
<span><span class="co">#&gt;  [99] digest_0.6.34               BiocGenerics_0.48.1        </span></span>
<span><span class="co">#&gt; [101] SparseArray_1.2.4           farver_2.1.1               </span></span>
<span><span class="co">#&gt; [103] rjson_0.2.21                memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [105] htmltools_0.5.7             pkgdown_2.0.7              </span></span>
<span><span class="co">#&gt; [107] lifecycle_1.0.4             GlobalOptions_0.1.2</span></span></code></pre></div>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Vipul Singhal, Joseph Lee.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
